specs/waku/mailserver.md

```diff
@@ -92,6 +93,63 @@ payload = "[" request-id last-envelope-hash [ cursor ] "]"
 
 If `Cursor` is not empty, it means that not all messages were sent due to the set `Limit` in the request. One or more consecutive requests MAY be sent with `Cursor` field filled in in order to receive the rest of the messages.
 
+### REST API
+
+In order to provide a stateless method of recovering messages from a mailserver, we define a simple REST api that SHOULD be used by clients.
+
+To do this we define a single endpoint that mailserver nodes MUST implement.
+
+#### GET `/v1/envelopes`
+
+This endpoint returns a list of `envelopes` for a given query.
+
+##### Parameters
+
+- **from [int]** - `required` - UNIX time in seconds; oldest requested envelope's creation time.
+- **to [int]** - `required` - UNIX time in seconds; newest requested envelope's creation time.
+- **bloom [string]** - `optional` - array of Waku topics encoded in a bloom filter to filter envelopes.
+- **limit [int]** - `required default: 100` - The maximum amount of envelopes to return.
+- **page [int]** - `optional` `default: 0` - The page to return.
```

Comment:
Here we want likely to use `cursor`  based pagination instead of `page` based, as page based is not stable.

Essentially mailserver will return a `cursor`(string) which is returned in the response, and used to fetch the next "page"

Reply:
https://jsonapi.org/profiles/ethanresnick/cursor-pagination/ for reference, the current mailserver implementation already uses a cursor

Reply:
good point, thanks @cammellos

Reply:
changed @cammellos